nix/pkgs.nix

```diff
@@ -0,0 +1,240 @@
+      })
+      # cross compilation logic
+      ({ pkgs, buildModules, config, lib, ... }:
+      let
+        buildFlags = map (opt: "--ghc-option=" + opt) [
+          "-fexternal-interpreter"
+          "-pgmi" "${config.hsPkgs.buildPackages.iserv-proxy.components.exes.iserv-proxy}/bin/iserv-proxy"
+          "-opti" "127.0.0.1" "-opti" "$PORT"
+          # TODO: this should be automatically injected based on the extraLibrary.
+          "-L${pkgs.windows.mingw_w64_pthreads}/lib"
+          "-L${pkgs.gmp}/lib"
+        ];
+        preBuild = ''
+          # unset the configureFlags.
+          # configure should have run already
+          # without restting it, wine might fail
+          # due to a too large environment.
+          unset configureFlags
+          PORT=$((5000 + $RANDOM % 5000))
+          echo "---> Starting remote-iserv on port $PORT"
+          WINEDLLOVERRIDES="winemac.drv=d" WINEDEBUG=-all+error WINEPREFIX=$TMP ${pkgs.buildPackages.winePackages.minimal}/bin/wine64 ${packages.remote-iserv.components.exes.remote-iserv}/bin/remote-iserv.exe tmp $PORT &
+          echo "---| remote-iserv should have started on $PORT"
+          RISERV_PID=$!
+        '';
+        postBuild = ''
+          echo "---> killing remote-iserv..."
+          kill $RISERV_PID
+        '';
+
+        testFlags = [ "--test-wrapper ${wineTestWrapper}/bin/test-wrapper" ];
+        wineTestWrapper = pkgs'.writeScriptBin "test-wrapper" ''
+          #!${pkgs'.stdenv.shell}
+          set -euo pipefail
+          WINEDLLOVERRIDES="winemac.drv=d" WINEDEBUG=-all+error LC_ALL=en_US.UTF-8 WINEPREFIX=$TMP ${pkgs.buildPackages.winePackages.minimal}/bin/wine64 $@*
+        '';
+        preCheck = ''
+          echo "================================================================================"
+          echo "RUNNING TESTS for $name via wine64"
+          echo "================================================================================"
+          cp ${pkgs.rocksdb}/lib/*.dll .
+          cp ${pkgs.openssl.bin}/bin/*.dll .
+          # copy all .dlls into the local directory.
+          # we ask ghc-pkg for *all* dynamic-library-dirs and then iterate over the unique set
+          # to copy over dlls as needed.
+          for libdir in $(ghc-pkg --package-db=$packageConfDir field "*" dynamic-library-dirs --simple-output|xargs|sed 's/ /\n/g'|sort -u); do
+            if [ -d "$libdir" ]; then
+              for lib in "$libdir"/*.{DLL,dll}; do
+                cp "$lib" .
+              done
+            fi
+          done
+        '';
+        postCheck = ''
+          echo "================================================================================"
+          echo "END RUNNING TESTS"
+          echo "================================================================================"
+        '';
+        withTH = { setupBuildFlags = buildFlags; setupTestFlags = testFlags;
```

Comment:
Is it possible to declutter by moving withTH and associated variables into a separate nix file?

Reply:
Let's see if this works.